tests/test_game_state.py

```
@@ -0,0 +1,367 @@
+    # Move piece to edge where rotation would be invalid
+    game_state.active_piece.x = settings.dimensions.BOARD_WIDTH - 1
+    game_state.active_piece.y = settings.dimensions.BOARD_HEIGHT - 1
+
+    # Try to rotate (might be invalid due to collision)
+    success = game_state.rotate_piece(clockwise=True)
+    # Result depends on piece type and position
+
+
+def test_hard_drop(game_state):
+    """Test hard drop functionality."""
+    original_y = game_state.active_piece.y
```

Ruff (0.14.2)

Local variable `original_y` is assigned to but never used

Remove assignment to unused variable `original_y`

(F841)